app/main.py

```diff
@@ -1,2 +1,51 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.size = 8
+        self.before_resize = int(self.size * 2 / 3)
+        self.hash_table = [[] for _ in range(self.size)]
+        self.length = 0
+
+    def __setitem__(self,
+                    key: (int, float, str, bool, tuple),
+                    value: Any) -> None:
+        if self.length == self.before_resize:
+            self.resize()
+        hashed_key = hash(key)
+        index = hashed_key % self.size
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, hashed_key, value]
+                self.length += 1
+                break
+
+            if self.hash_table[index][0] == key and \
+                    self.hash_table[index][1] == hashed_key:
+                self.hash_table[index][2] = value
+                break
+            index = (index + 1) % self.size
+
+    def __getitem__(self, key: (int, float, str, bool, tuple)) -> Any:
```

Comment:
```suggestion
    def __getitem__(self, key: Hashable) -> Any:
```
[Hashable](https://docs.python.org/3/library/typing.html#typing.Hashable)